docs/synapse/userguides/storm_adv_functions.rstorm

```diff
@@ -0,0 +1,708 @@
+      $initMetaSource()
+      
+      //Yield the meta:source node into this function's pipeline
+      yield $srcnode
+      
+      //Set the $date variable to the value of the node data key mysource:report:date from
+      // the meta:source node.
+      $date = $node.data.get(mysource:report:date)
+      
+      //If there is no value for this key return the integer 0
+      if ($date = $lib.null) { return((0)) }
+      
+      //Otherwise return the date
+      return($date)
+  }
+  
+  //Assign the value returned by this function to the variable $date for use by the
+  // invoking Storm pipeline. This value can be passed to another function that retrieves
+  // the latest feed data.
+  $date = $getLastReportDate()
+ 
+
+.. _storm-func-emitter:
+
+Data Emitter Functions
+++++++++++++++++++++++
+
+**Data emitter** functions emit data using the ``emit`` keyword. The ``stop`` keyword can optionally
+be used to halt processing and exit the function. The ``emit`` keyword **must** be present for a data
+emitter function to execute properly.
+
+Data emitter functions **stream** data (technically, they return a **generator object** that is iterated
+over). They are designed to emit data as it is available while continuing to execute. They may be invoked
+with **for** or **while** loops for this purpose.
+
+Use Cases
+*********
+
+Data emitter functions can be used to:
+
+- Consume data from sources that paginate results, where you want to mask the pagination (i.e., a data
+  emitter can consume and emit the first page of results; then consume and emit the next page; and so on).
+- Consume data from sources that stream results, where the data emitter is used to continue the streaming
+  behavior.
+
+.. TIP::
+  
+  Data emitters can be used to emit nodes (e.g., ``emit $node``), though this is an uncommon use case.
+  The ability of data emitters to emit data incrementally is useful when consuming large result sets from
+  an API. "Subsets" of results (such as individual JSON objects from a JSON blob) can be made available
+  more quickly (e.g., to another function responsible for creating nodes from the JSON) while the
+  emitter continues to process data.
+  
+  In contrast, if the same set of API results was consumed by a callable function, the function would
+  need to consume the entire result set before returning.
+
+Pseudocode
+**********
+
+::
+  
+  function data_emitter() {
+      
+      for $thing in $things {
+          <do stuff>
+          emit $thing
+      }
+  }
+
+Or:
+
+::
+  
+  function data_emitter() {
+      
+      for $thing in $things {
+          <do stuff>
+          emit $thing
+          
+          if ($thing = "badthing") {
+              stop
+          }
+      }
+  }
+
+Or:
+
+::
+  
+  function data_emitter() {
+  
+      while (1) {
+        <do stuff>
+        emit $thing
+        
+        if (<end condition>) { stop }
+      }
+      
+      <update something to contine while loop>
```

Comment:
this looks like maybe it should be in loop? (also "contine")

Reply:
fixed typo, moved pseudocode into 'while' loop